fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/api/networking/v1/PayloadTypeRegistry.java

```diff
@@ -43,6 +43,39 @@ public interface PayloadTypeRegistry<B extends PacketByteBuf> {
 	 */
 	<T extends CustomPayload> CustomPayload.Type<? super B, T> register(CustomPayload.Id<T> id, PacketCodec<? super B, T> codec);
 
+	/**
+	 * Registers a splittable custom payload type.
+	 *
+	 * <p>This must be done on both the sending and receiving side, usually during mod initialization
+	 * and <strong>before registering a packet handler</strong>.
+	 *
+	 * <p>Payload types registered with this method will be split into multiple packets,
+	 * if it ends up exceeding vanilla size limit.
+	 *
+	 * @param id    the id of the payload type
+	 * @param codec the codec for the payload type
+	 * @param <T>   the payload type
+	 * @return the registered payload type
+	 */
+	<T extends CustomPayload> CustomPayload.Type<? super B, T> registerSplittable(CustomPayload.Id<T> id, PacketCodec<? super B, T> codec);
```

Comment:
Is `registerSplittable` a good name for this? Should the modder care that the packet is split, or would something more generic such as `registerLarge` be better? The splitting is likely an impl detail they dont need to care about.